collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java

```diff
@@ -164,19 +156,34 @@ public void openHierarchy() {
         formController.getAuditEventLogger().logEvent(AuditEvent.AuditEventType.HIERARCHY, true, clock.getCurrentTime());
     }
 
+    public void logFormEvent(String event) {
```

Comment:
Rather than having to pass the (or grab statically) form identifier hash around it seemed nice to just encapsulate the logging on the view model here. I think we'd want a `FormEventLogger` interface if we were logging inside a widget but I didn't run into that.